applications/ShapeOptimizationApplication/python_scripts/design_logger_gid.py

```diff
@@ -26,43 +26,57 @@
 class DesignLoggerGID( DesignLogger ):
 
     # --------------------------------------------------------------------------
-    def __init__( self, ModelPartController, OptimizationSettings ):
-        self.OptimizationModelPart = ModelPartController.GetOptimizationModelPart()
-        self.DesignSurface = ModelPartController.GetDesignSurface()
-        self.OutputSettings = OptimizationSettings["output"]
+    def __init__( self, model_part_controller, optimization_settings ):
+        self.output_settings = optimization_settings["output"]
+        default_gid_settings = Parameters("""
+        {
+            "name"              : "gid",
+            "gid_configuration" : {
+                "result_file_configuration" : {
+                    "gidpost_flags"         : {
+                        "GiDPostMode"       : "GiD_PostBinary"
+                    },
+                    "output_frequency": 1.0
+                }
+            }
+        }""")
+        self.output_settings["output_format"].RecursivelyValidateAndAssignDefaults(default_gid_settings)
```

Comment:
If you do recursive validation the you cannot add entries to e.g. `gidpost_flags`

Reply:
I stumbled upon this validation thing already quite some times already.
The missing feature of the parameters is to allow more settings than there are in the defaults, and just compare and validate what is given in the defaults...
Is there another way of doing such a thing?

Reply:
Allowing more things in there would beat the idea of allowing only existing parameters 

The proper solution would be to not add any validation -> the validation should only be done in the class that knows abt the defaults
Basically what you are doing here is redundant that breaks as soon as the parameters in the class change